src/utils/source-search.js

```diff
@@ -20,28 +17,19 @@ function SearchState() {
  * @memberof utils/source-search
  * @static
  */
-function getSearchState(cm) {
+function getSearchState(cm: any) {
   return cm.state.search || (cm.state.search = new SearchState());
 }
 
 /**
  * @memberof utils/source-search
  * @static
  */
-function getSearchCursor(cm, query, pos) {
-  // If the query string is all lowercase, do a case insensitive search.
-  return cm.getSearchCursor(query, pos,
-    typeof query == "string" && query == query.toLowerCase());
-}
+function getSearchCursor(cm, query: string, pos, modifiers: SearchModifiers) {
+  const { caseSensitive } = modifiers;
 
-/**
- * Ignore doing outline matches for less than 3 whitespaces
- *
- * @memberof utils/source-search
- * @static
- */
-function ignoreWhiteSpace(str) {
-  return /^\s{0,2}$/.test(str) ? "(?!\s*.*)" : str;
+  const regexQuery = buildQuery(query, modifiers, { isGlobal: false });
+  return cm.getSearchCursor(regexQuery, pos, !caseSensitive);
```

Comment:
The case sensitive flag is only useful with `cm.getSearchCursor` when the query is a string, if it's a regex then it's an ignored argument.